app/route53.py

```diff
@@ -0,0 +1,248 @@
+
+    change = {
+        'Action': 'DELETE',
+        'ResourceRecordSet': {
+            'Name': cname_domain,
+            'Type': 'CNAME',
+            'ResourceRecords': record['ResourceRecords']
+        }
+    }
+    if 'TTL' in record:
+        change['ResourceRecordSet']['TTL'] = record['TTL']
+    if 'SetIdentifier' in record:
+        change['ResourceRecordSet']['SetIdentifier'] = record['SetIdentifier']
+
+    change_batch = {
+        'Comment': 'DELETE {0}'.format(cname_domain),
+        'Changes': [change],
+    }
+    log.info(pformat(change_batch))
+
+    r = client.change_resource_record_sets(
+        HostedZoneId=zone_id,
+        ChangeBatch=change_batch)
+    log.info(r)
+    if r['ResponseMetadata']['HTTPStatusCode'] != 200:
+        msg = 'delete_cname failed with:\n' + pformat(change)
+        raise Route53Error(msg)
+
+
+def _get_zone_id(client, url):
+    """Get the ID of the Hosted Zone that services this `url`.
+
+    Parameters
+    ---------
+    client :
+        Boto3 Route 53 client.
+    url : str
+        A URL. :func:`_derive_fully_specified_domain` is used to turn this into
+        a fully specified domain.
+
+    Returns
+    -------
+    zone_id : str
+        Route 53 Hosted Zone ID that services the domain of the given URL.
+    """
+    zones = client.list_hosted_zones()
+
+    fsd = _derive_fully_specified_domain(url)
+    zone_id = None
+    for z in zones['HostedZones']:
+        if fsd == z['Name']:
+            zone_id = z['Id']
+
+    if zone_id is None:
+        msg = 'Could not find hosted zone for fully specified ' \
+              'domain: {0}'.format(fsd)
+        log.error(msg)
+        log.error(pformat(zones))
+        raise Route53Error(msg)
+
+    log.info('Using HostedZoneId: {0}'.format(zone_id))
+    return zone_id
+
+
+def _find_cname_record(client, zone_id, cname_domain):
+    """Find an existing record for the `cname_domain`, or `None` if one does
+    not exist.
+
+    Parameters
+    ----------
+    client :
+        Boto3 Route 53 client.
+    zone_id : str
+        The Hosted Zone ID for the CNAME.
+    cname_domain : str
+        The CNAME domain. See
+        http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/ResourceRecordTypes.html#CNAMEFormat
+
+    Returns
+    -------
+    record : dict
+        The ``ResourceRecordSet`` with a ``Name`` corresponding to the
+        `cname_domain`. E.g.,
+
+        .. code-block:: python
+
+           [{'Name': 'www.mydomain.org.',
+             'ResourceRecords': [{'Value': 'origin_domain.com'}],
+             'TTL': 14400,
+             'Type': 'CNAME'}]
+
+        `None` is return is not ResourceRecordSet matching `cname_domain` is
+        found.
+    """
+    # turns out boto3 doesn't need StardRecordName in lexicographic order
+    # despite their docs.
+    # name = _lexicographic_order_domain(cname_url)
+
+    if not cname_domain.endswith('.'):
+        cname_domain = cname_domain + '.'
```

Comment:
It might be worth creating a utility method to munge domain names into the pedantically qualified form and reusing it in other methods.